sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryFragment.java

```diff
@@ -37,9 +37,13 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 	}
 
 	@Override
-	public void onDestroyView() {
-		super.onDestroyView();
-		orderHistoryPresenter.onDetach();
+	public void onDestroy() {
+		if (orderHistoryPresenter != null) {
+			orderHistoryPresenter.onDetach();
+			orderHistoryPresenter = null;
+		}
+		orderHistoryRecyclerAdapter = null;
```

Comment:
protected orderHistoryRecyclerAdapter in other places - related methods could be called from async operation even after onDestroy

Reply:
and why you need to null out adapter?

Reply:
I think the approach should be to null out the view in base presenter, and check in each presenter for a null view before each operation.

Reply:
1. the adapter contains the recyclerview the contains the activity as a context, so in some scenario is can be leaked.
2. it is like that but it doesn't mean that something else does not save the activity instance.